### ISSUE TITLE
When creating a guide the user should be able to skip the stages bit

### ISSUE THREAD

**Issue by simonv3** [openfarmcc/OpenFarm#813]
The idea being that we enable blog post style guides, which we would then encourage users to turn into more detailed posts later, or get people to fork and build their own guides based on it.

**Comment by CloCkWeRX** [comment]
I like that - when I went through the experience, I kind of was wishing it was a bit more like wikihow. IE that I could add a method or particular tip to someone else's guide.

The onus of sharing my specific method in 100% structured detail is quite intimidating - a tip or comment first is easy, then maybe a formatted post for longer content.

If the system then started looking at the text I entered and asking me structured questions; that would make a lot of sense.
At its simplest: I put in a keyword like 'water', the system asks if I want to put in watering instructions (frequency, amount, etc).

**Comment by simonv3** [comment]
Oooh, I like that a lot. 

In @roryaronson's original sketches for what a guide would look like comments were a definite part. Maybe we should push those forward in priority? After talking to @sophiakc (on Slack, I don't think she's on GitHub) she suggested that we activate more "social" interaction on the site as well.

**Comment by simonv3** [comment]
This would probably be affected by #847

**Comment by sophiakc** [comment]
Agree with that @simonv3 @CloCkWeRX !

@roryaronson @simonv3 @CloCkWeRX For the guide creation, would you be open to try something radically new, putting on a side the actual structure of the guide creation flow and offering a way to simply add a title, some text and structured elements like we could do with the following UI:
![screencapture-docs-google-forms-d-1nviqdrmufvutlwa3naowtwykcsv6aprqooqo1vhdfwi-edit-1487717483880](https://cloud.githubusercontent.com/assets/3863675/23188961/8ece059a-f845-11e6-8d6b-f049480437d1.png)
![screencapture-sites-google-s-0b4a1xw3oy2hbazlkwkx2d3faoxc-p-0b4a1xw3oy2hbtm94q3pqanrrr28-edit-1487717513733](https://cloud.githubusercontent.com/assets/3863675/23188962/8ed42f74-f845-11e6-9bb7-1e3b0f5b2159.png)

The structured elements could be:
- specific stages
- tips, comments
- highlights words to annotate
- tags
- links
- open questions

**Comment by simonv3** [comment]
@sophiakc we had a less sophisticated version of this originally where people were just shown a blank guide and asked to fill it in. The edit and create pages used to be really similar. 

I think you're suggesting a shorter create process where the user fills in the things you mentioned and then gets shown the full guide, where they can fill in details like stages, etc?

**Comment by sophiakc** [comment]
@simonv3 "shorter create process" yes! 
+ adding some pre-defined structured sections, meant for the API and having harmonized way of structuring a guide for users

Advantages: 
- shorten the process at a minimum level of requirements
- make it playful by adding myself the pre-defined sections
- make it collaborative, build upon each other inputs: someone else can add tags, highlights specific portions of text to say for instance: this is prep soil related, or this is usually done during fall...

**Comment by simonv3** [comment]
Oh interesting about that collaborative aspect. How do you see that happening? Anyone can edit a guide belonging to a user? How would a user feel about their guide being edited by others? Or would there be restrictions? 

I'd love it if you could elaborate on that a bit more?

**Comment by sophiakc** [comment]
@simonv3 
It would be very similar to [Medium](medium.com) UI = the author is the only one having edit access. And others can highlights, add tags..., and so forth upon what the author has written.
So then later, the author or others can add more "qualitative structure"

does it make sense?

**Comment by simonv3** [comment]
Yeah that makes sense!

**Comment by sophiakc** [comment]
@roryaronson @walkah @CloCkWeRX what's your thoughts on that?

**Comment by roryaronson** [comment]
I think encouraging readers to annotate/add to/interact with the guide is a great idea. Though I think the Guide must remain authored by a single person as we seem in agreement on. Comments on the guide (as a whole) I think would be the most logical first step. Eventually comments could be linked to a specific section of the guide, or something. And we could also look into implementing a "suggested edits" feature where someone could suggest certain additions/changes to the guide that the author could approve or not. For example: suggesting a tag, or even a whole new life stage if we got really ambitious. But again, these suggestions must be controlled by the original single author otherwise we would encourage guide gentrification (all ideas flow into canonical behemoth guides)

**Comment by sophiakc** [comment]
Hey @roryaronson 
Not sure about what you're answering to...
My initial question is: For the guide creation, would you be open to try something radically new, putting on a side the actual structure of the guide creation flow and offering a way to simply add a title, some text and structured elements like we could do with the above UI? 
If yes, I'll prep the wires for the new guide creation flow.

Yes, one single author.
Multiple contributors to structure the content, like with Medium.
So the advantage is: instead of having a very structured, complex guide to fill in, as a user, I am free to write whatever I want. And after/later, this content can be structured (to have the same structure for all guides), either by myself as the guide author, or by others (like wikipedia, Medium...)

**Comment by sophiakc** [comment]
Initial title of this issue: When creating a guide the user should be able to skip the stages bit

**Comment by sophiakc** [comment]
@roryaronson what's your take on that?

**Comment by andru** [comment]
> My initial question is: For the guide creation, would you be open to try something radically new, putting on a side the actual structure of the guide creation flow and offering a way to simply add a title, some text and structured elements like we could do with the above UI?

I think there's awesome potential to strike gold here @sophiakc :)

If we can make the addition of the phases feel creative rather than laborious, it could really help reduce the feeling of intimidation and suppression of creativity that comes over when you're given a form to fill out which doesn't let you colour outside the lines.

If we can help the user feel like they're writing a story and yet guide their hand with structure as they go, we could reduce that intimidation. 

I love the ideas of being able to highlight and inline-comment on guides like you can on Medium. We could bring that value very easily by integrating [Genius](https://genius.com/web-annotator) if it's not something we had the time to build ourselves. It would mean an third party sign up for the user to annotate, though, which sucks.

I think my dream for the authoring UI would be if we had NLP smart enough to extract data as the user writes their guide in natural language, building the structured data it in a sidebar as the user writes, and giving them the option of tweaking it before they publish... 
e.g. the user writes "plant out after 4 weeks or when plants have their third set of leaves" and we can actually turn that into data: [plant out, time:4 weeks], [plant out, maturity:3rd leaves]

But that is *complex*. I played with doing something similar to that for Hortomatic with much more structured vocab and it even there the edge cases were surprisingly tricky. 

What @CloCkWeRX suggested seems like a great approach without adding too much complexity:

>At its simplest: I put in a keyword like 'water', the system asks if I want to put in watering instructions (frequency, amount, etc).

Although there's the possibility for a lot of false-positives without a bit of NLP to only extract the verbs... e.g. water, plant, spray, compost (etc) are each both noun and verb.

**Comment by roryaronson** [comment]
I like the idea of Guide creation being fast and easy for someone and less intimidating and laborious. Right now the flow has the user go through a pretty structured process from the start, which I think is good for an expert who has all the needed knowledge/photos/etc now and wants to create a complete guide. A less structured process that becomes structured later would be better for someone who maybe doesn't have all the info right now and is going to make more of a work-in-progress guide.

Perhaps combining the Guide creation process into the Guide editing process could satisfy both types of authors? So rather than there being a multi-page Guide creation *flow* that the author must go through from start to finish, we could start the author off on the Guide itself and walk them through adding the pieces (overview, life stages, etc) one by one in more of a choose-your-own-adventure style?

**Comment by sophiakc** [comment]
@andru interesting input! I'm open to discuss how this could look like UI-wise regarding what's doable and chosen on the technical side.

@roryaronson where can I have access to metrics such as:
- how many guides have been created and when
- among them how many are complete/ incomplete
- what are the pages/ sections that are not completed
- which sections are useful for Farmbot?

@roryaronson I am not sure I get your explanation about combining the two? can you rephrase it?

**Comment by roryaronson** [comment]
Here are some numbers from the admin side of the site:
<img width="1052" alt="screen shot 2017-03-08 at 7 39 46 pm" src="https://cloud.githubusercontent.com/assets/5524043/23734917/1cc91cde-0437-11e7-96f7-a3069d436d83.png">

You can get a feel for "completeness" of guides by just browsing around and see what people have created and what is missing.

FarmBot currently uses Crop photos and information (the structured, factual content about the Crops). We are also working on Crop icons which we upload to OF once the infrastructure is in place in a week or so. We would *like* to use Guide information, but it must be very structured and complete for it to be useful to FarmBot, and we're a long ways off from having that on OpenFarm and also the software to use it on the FarmBot side of things.

For the explanation of the two. I was referring to combining the guide *creation* flow with the guide *editing* screen. Guide creation happens in a multi-page flow. Guide editing happens inline on the Guide page itself in a WYSIWYG fashion. To combine the two, we *could* do away with the guide creation flow and instead modify the inline editor to be not only be for editing, but also a semi-guided process for creating the guide from scratch. It could simplify stuff so that the creation and editing works for both types of authors: someone who wants something less structured and less complete, vs someone who has more info and wants to go fill everything out. Just an idea

**Comment by sophiakc** [comment]
Hey @roryaronson

Thx for providing the admin document.
Can you legend what does the different items of model name exactly correspond to? Are they part of the user flow for a given user?
- Regarding guides, there are 170 created right? Is there a way to get more details of when they were created, by who, location...?
- Regarding impressions, is it guide pdf impression? guide views?
- is there a way to know who used the guide edition feature?

Regarding the guide creation
- so right now, only the crop data and none of the guide data itself are used by Farmbot? 
- How do you see the "semi-guided process for creating the guide from scratch"?
- Why in the first place did you pick two different approaches for guide creation vs. edition?
- if the two approaches are combined into the guide edition UI, would you consider as a try an even more simple version of it: the one I was suggesting above?
